arch/Kconfig

```diff
@@ -83,6 +83,15 @@ config BIG_ENDIAN
          is used to select linker script OUTPUT_FORMAT and command
          line option for gen_isr_tables.py.
 
+config 64BIT
+	bool
+	help
+	  This option tells the build system that the target system is
+	  using a 64-bit address space, meaning that pointer and long types
+	  are 64 bits wide. This option is selected by arch/$ARCH/Kconfig,
+	  soc/**/Kconfig, or boards/**/Kconfig and the user should generally
+	  avoid modifying it.
```

Comment:
This gets a little murky with x86_64 hardware, where the use of the x32 ABI is a legitimate optimization choice and probably does want to be under the control of the app.  I'd suggest just dropping the last sentence, or alternatively I could make x32 tuning a separate kconfig that transitively selects this one (but then inevitably someone is going to ask why native_posix and x86_64 use different tunables for the same purpose).

Reply:
Having this option makes a lot of sense. But I disagree with the way you are describing and defining it.
* You are defining it without a prompt, so a "user" won't be able to modify it.
* The fact that the target HW (like for the POSIX arch) supports x64 does not mean that the intention is to compile the application as x64. (Indeed, in the case of @andyross x86_64, one can indeed aim at x32 for savings) But in the case of the POSIX arch (or native_posix), the most logical target I would see is x86, not x64:
  * Zephyr targets constrained embedded systems, which are rarely 64bit.
  * The POSIX arch is a tool to have a nicer environment to run and debug the code you would otherwise in an embedded target. So in this situation there would be no benefit of compiling as x64, only drawbacks as a you would see a different subset of bugs.
  * Having the default as x64 would break any application out there which is not 64 bit compliant. Effectively limiting the use of the POSIX arch.

So, if you want to define it as this level, I'd think that at least you need to add a prompt to it, so it can be changed by the user.
(The default for the POSIX arch or native_posix cannot be to enable this setting. But that does not mean we should prevent a user/application from doing so)